logdna/client.go

```diff
@@ -62,8 +63,38 @@ type channelResponse struct {
 	URL             string            `json:"url,omitempty"`
 }
 
-// MakeRequestView makes a request to the config-api and parses and returns the response
-func MakeRequestView(c *Client, url string, urlsuffix string, method string, payload viewPayload) (string, error) {
+// MakeRequestAlert makes a HTTP request to the config-api with alert payload data and parses and returns the response
+func MakeRequestAlert(c *Client, url string, urlsuffix string, method string, payload ViewPayload) (string, error) {
+	pbytes, err := json.Marshal(payload)
+	if err != nil {
+		return "", err
+	}
+	req, err := http.NewRequest(method, url+urlsuffix, bytes.NewBuffer(pbytes))
+	if err != nil {
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("servicekey", c.ServiceKey)
+	resp, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return "", fmt.Errorf(`Error with alert: %s`, err)
+	}
+	defer resp.Body.Close()
+	var result alertResponsePayload
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode != 200 {
+		return "", errors.New(result.Error)
+	}
+
+	return result.Presetid, nil
```

Comment:
this field should be `PresetID`, or at least all `ID` fields should be consistently cased (`AlertID`, `Presetid`, `Viewid`)